analysis_driver/quality_control/contamination_blast.py

```diff
@@ -0,0 +1,107 @@
+import os
+from ete3 import NCBITaxa
+from egcg_core import executor
+from .quality_control_base import QualityControl
+from analysis_driver.config import default as cfg
+
+class ContaminationBlast(QualityControl):
+    def __init__(self, dataset, working_dir, fastq_files):
+        super().__init__(dataset, working_dir)
+        self.working_dir = working_dir
+        self.fastq_files = fastq_files
+
+    def sample_fastq_command(self, fastq_file):
+        seqtk_bin = cfg['tools']['seqtk']
+        fasta_outfile = os.path.join(self.working_dir, self.dataset.name, fastq_file.split('.')[0] + '_sample3000.fasta')
+        seqtk_sample_cmd = '%s sample %s 3000 | seqtk seq -a > %s' % (seqtk_bin, fastq_file, fasta_outfile)
+        return seqtk_sample_cmd, fasta_outfile
+
+    def fasta_blast_command(self, fasta_file):
+        blastn_bin = cfg['tools']['blastn']
+        nt_db = cfg['contamination-check']['nt_db']
+        blast_outfile = fasta_file.split('.')[0] + '_blastn'
+        blast_cmd = "%s -query %s -db %s -out %s -outfmt '6 qseqid sseqid length pident evalue sgi sacc staxids sscinames scomnames stitle'" % (blastn_bin, fasta_file, nt_db, blast_outfile)
```

Comment:
Is the blast command meant to be multithreaded?
The process has been given 2 CPU but could probably be given more.